keps/sig-network/3705-cloud-node-ips/README.md

```diff
@@ -477,31 +543,46 @@ issues.
 
 ###### Does enabling the feature change any default behavior?
 
-No
+Kubelet will begin setting two annotations on each node rather than
+one. However, cloud providers will still prefer the old annotation and
+will only read the new annotation if the old one isn't there.
+
+So assuming no bugs, the only visible change after (just) enabling the
+feature is the duplicate annotation.
 
 ###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
 
 Yes, as long as you also roll back the kubelet configuration to no
 longer use the new feature.
 
+One edge case: if a user who was not previously setting `--node-ip`
+upgrades, enables the feature, starts using `--node-ip`, and then
+rolls back the kubelet and its configuration but _not_ the cloud
+provider, then they will end up with a node that has a stale "new"
+annotation (with the `--node-ip` value they had been using while
+upgraded), but no "old" annotation (because the rolled-back kubelet
+configuration no longer specifies `--node-ip` so kubelet will delete
+the old annotation). The (not-rolled-back) cloud provider will then
+mistake this as being the future rather than the past, assuming
+kubelet intentionally set only the new annotation and not the old
+annotation, and so it will obey the (stale) new annotation.
+
+That could be prevented by stretching out the feature enablement over
+a few more releases, and not having the cloud-provider switch from
+"always use the old annotation" to "prefer the old annotation when
+it's set but use the new annotation if it's the only one" until a few
+releases later. If we were going to do that, then perhaps we should
+split the annotation renaming into a separate KEP from the dual-stack
+`--node-ip` handling, so that the edge cases of the annotation
+renaming don't have to slow down the dual-stack `--node-ip` adoption.
+
 ###### What happens if we reenable the feature if it was previously rolled back?
 
 It works.
 
 ###### Are there any tests for feature enablement/disablement?
 
-<!--
-The e2e framework does not currently support enabling or disabling feature
-gates. However, unit tests in each component dealing with managing data, created
-with and without the feature, are necessary. At the very least, think about
-conversion tests if API types are being modified.
-
-Additionally, for features that are introducing a new API field, unit tests that
-are exercising the `switch` of feature gate itself (what happens if I disable a
-feature gate after having objects written with the new field) are also critical.
-You can take a look at one potential example of such test in:
-https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
--->
+No, but enablement/disablement alone should not change any behavior.
```

Comment:
I think there is a minor bit of persistent state which is the annotation.

So having a test that effectively checks if the annotations will be cleared would be useful
(it can be a test that already injects a node with annotation and starting kubelet (unit test) and checking if that annotation would be clearered
[Assuming we will apply the comment from Tim above.]

Would that make sense?